conformance/lang/expressions/mapping_constructor_expr.balt

```diff
@@ -0,0 +1,1882 @@
+
+    MyRecord1|record {|int otherField;|} h = {myField1: 776777777, myField2: true};
+    io:println(h is MyRecord1); // @output true
+
+    record {boolean field3;}|record {||} i = {"field1": 0, "field2": "mouse", field3: false};
+    io:println(i is record {| boolean field3; anydata...; |}); // @output true
+
+    MyRecord2|MyRecord1 o = {myField1: "776777777", myField2: true, "other": false};
+    io:println(o is MyRecord2); // @output true
+
+    UnionRecord p = {myField1: "776777777", myField2: true, "other": false};
+    io:println(p is MyRecord2); // @output true
+}
+
+Test-Case: error
+Description: Test if the applicable contextually expected type is a union type descriptor, then any members of the 
+             union that are inconsistent with the field names specified in a specific-field in the mapping-constructor-expr 
+             will be ignored; it is a compile-time error if this does not leave a single mapping type descriptor.
+Fail-Issue: ballerina-platform/ballerina-lang#33056
+Labels: mapping-constructor-expr, map-type, module-type-defn, record-rest-descriptor, record-type, 
+       record-type-inclusion, string, union-type
+
+type MyRecord1 record {|
+    int myField1;
+    boolean myField2;
+|};
+
+type MyRecord2 record {
+    string myField1;
+    boolean myField2;
+};
+
+type UnionRecord MyRecord1|MyRecord2;
+
+function errorFunction() {
+    map<int>|map<boolean> _ = {field1: 1, field2: 2}; // @error remaining type is a union
+
+    map<int>|map<boolean> _ = {field1: true, field2: false}; // @error remaining type is a union
+
+    map<string>|record {|boolean fieldA1; string fieldA2;|} _ = {fieldA1: "String1", fieldA2: "String2"}; // @error remaining type is a union
+
+    record {|int fieldA; int fieldB;|}|record {|string fieldA; int fieldB;|} _ = {fieldA: "String1", fieldB: 44}; // @error remaining type is a union
+
+    record {|int field1; string field2; boolean field3?;|}|record {|int field1; int field2;|} _ = {field1: 0, field2: "mouse"}; // @error remaining type is a union
+
+    record {|string field2; boolean...; |}|record {|string field2; int...; |} _ = {field2: "Car", "field3": false}; // @error remaining type is a union
+    
+    record {string field2;}|record {int field2;} _ = {field2: "Car", "field3": false}; // @error remaining type is a union
+
+    record {|int field1; *MyRecord1;|}|record {|int field1; int myField1; int myField2;|} _ = {field1: 43, myField1: 3443, myField2: false}; // @error remaining type is a union
+
+    MyRecord1|record {|int myField1; int myField2;|} _ = {myField1: 776777777, myField2: true}; // @error remaining type is a union
+
+    MyRecord2|MyRecord1 _ = {myField1: "776777777", myField2: true}; // @error remaining type is a union
+
+    UnionRecord _ = {myField1: "776777777", myField2: true}; // @error remaining type is a union
+}
+
+Test-Case: error
+Description: Test if the applicable contextually expected type is a union type descriptor, then any members of the 
```

Comment:
Here we have tested only when there are multiple valid type descriptors in the union type for the mapping constructor. Shall we add few tests which don't contain any valid type descriptor in the union type for the mapping constructor?
Ex: 
`map<int>|map<byte> _ = {field1: "1", field2: "2"};`

Reply:
added in [a3ba756](https://github.com/ballerina-platform/ballerina-spec/pull/1047/commits/a3ba756463219f5c1e5b7095f194486a9028b0c9)